dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,851 @@
+                        .findContentletByIdentifier(identifier.getId(), mode.showLive, lid, APILocator.getUserAPI()
+                            .getSystemUser(), true);
+                    if (cc != null && UtilMethods.isSet(cc.getInode()) && !cc.isArchived()
+                            && LanguageWebAPI.canApplyToAllLanguages(cc)) {
+                        contentlet = cc;
+                    } else {
+                        CacheLocator.getVeloctyResourceCache()
+                            .addMiss(filePath);
+                        throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+                    }
+                }
+            }
+
+            Logger.debug(this, "DotResourceLoader:\tWriting out contentlet inode = " + contentlet.getInode());
+            return buildVelocity(contentlet, identifier, mode, filePath);
+
+        } catch (Exception e) {
+            throw new ResourceNotFoundException("Contentlet Velocity file not found:" + filePath);
+        }
+    }
+
+
+
+
+    @Override
+    public void invalidate(Object obj, PageMode mode) {
+        Contentlet asset = (Contentlet) obj;
+        CacheLocator.getContentletCache()
+            .remove(asset.getInode());
+
+        String folderPath = mode.name() + java.io.File.separator;
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        velocityRootPath += java.io.File.separator;
+
+        Set<Long> langs = new HashSet<Long>();
+        langs.add(asset.getLanguageId());
+        if (LanguageWebAPI.canApplyToAllLanguages(asset)) {
+            for (Language ll : APILocator.getLanguageAPI()
+                .getLanguages()) {
+                langs.add(ll.getId());
+            }
+        }
+        for (Long langId : langs) {
+            String filePath = folderPath + asset.getIdentifier() + "_" + langId + "." + VelocityType.CONTENT.fileExtension;
+            java.io.File f = new java.io.File(velocityRootPath + filePath);
+            f.delete();
+            DotResourceCache vc = CacheLocator.getVeloctyResourceCache();
+            vc.remove(ResourceManager.RESOURCE_TEMPLATE + filePath);
+        }
+        List<Field> fields;
+        try {
+            fields = asset.getContentType()
+                .fields();
+            for (Field field : fields) {
+                new FieldLoader().invalidate(field, mode);
+            }
+        } catch (DotDataException | DotSecurityException e) {
+            throw new DotStateException(e);
+        }
+
+
+        try {
+            if (asset.getContentType()
+                .baseType() == BaseContentType.HTMLPAGE) {
+                new PageLoader().invalidate(asset, mode);
+            }
+
+            if (asset != null && asset.isVanityUrl()) {
+                // remove from cache
+                VanityUrlServices.getInstance()
+                    .invalidateVanityUrl(asset);
+            }
+            if (asset != null && asset.isKeyValue()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)